vnext/Desktop.UnitTests/MemoryMappedBufferTests.cpp

```diff
@@ -0,0 +1,164 @@
+        GetLastError() != ERROR_FILE_NOT_FOUND) {
+      std::terminate();
+    }
+    if (PathFileExistsW(m_testFileName.c_str())) {
+      std::terminate();
+    }
+  }
+
+  TEST_METHOD_INITIALIZE(Initialize) {
+    DeleteTestFile();
+  }
+
+  TEST_METHOD_CLEANUP(CleanUp) {
+    DeleteTestFile();
+  }
+
+ public:
+  MemoryMappedBufferUnitTests() : m_testFileName{GetTestFileName()} {}
+
+  TEST_METHOD(SimpleTest_NoOffest) {
+    constexpr const char *const content = "This is a very interesting string.";
+    const size_t size = strlen(content);
+    WriteTestFile(content, size);
+
+    std::shared_ptr<Buffer> buffer =
+        MakeMemoryMappedBuffer(m_testFileName.c_str());
+
+    Assert::IsTrue(buffer->size() == size);
+    Assert::IsTrue(
+        strcmp(checkedReinterpretCast<const char *>(buffer->data()), content) ==
+        0);
+  }
+
+  TEST_METHOD(SimpleTest_WithOffest) {
+    constexpr const char *const fileContent =
+        "This is another very interesting string.";
+    const size_t fileSize = strlen(fileContent);
+    WriteTestFile(fileContent, fileSize);
+
+    const size_t fileOffset = 3;
+    std::shared_ptr<Buffer> buffer =
+        MakeMemoryMappedBuffer(m_testFileName.c_str(), fileOffset);
+
+    Assert::IsTrue(buffer->size() == fileSize - fileOffset);
+    Assert::IsTrue(
+        strcmp(
+            checkedReinterpretCast<const char *>(buffer->data()),
+            fileContent + fileOffset) == 0);
+  }
+
+  TEST_METHOD(EdgeCaseFileSizeTest_NoOffset) {
+    std::string content("a", GetPageSize());
+    WriteTestFile(content.c_str(), content.length());
+
+    std::shared_ptr<Buffer> buffer =
+        MakeMemoryMappedBuffer(m_testFileName.c_str());
+
+    Assert::IsTrue(buffer->size() == content.length());
+    Assert::IsTrue(
+        strcmp(
+            checkedReinterpretCast<const char *>(buffer->data()),
+            content.c_str()) == 0);
+  }
+
+  TEST_METHOD(EdgeCaseFileSizeTest_WithOffset) {
+    std::string content("a", GetPageSize());
+    WriteTestFile(content.c_str(), content.length());
+
+    const size_t fileOffset = 15;
+    std::shared_ptr<Buffer> buffer =
+        MakeMemoryMappedBuffer(m_testFileName.c_str(), fileOffset);
+
+    Assert::IsTrue(buffer->size() == content.length() - fileOffset);
+    Assert::IsTrue(
+        strcmp(
+            checkedReinterpretCast<const char *>(buffer->data()),
+            content.c_str() + fileOffset) == 0);
+  }
+
+  TEST_METHOD(ErrorTest_EmptyFile) {
+    WriteTestFile("", 0);
+
+    Assert::ExpectException<JSINativeException>([this] {
+      std::shared_ptr<Buffer> buffer =
+          MakeMemoryMappedBuffer(m_testFileName.c_str());
+    });
+  }
+
+  TEST_METHOD(ErrorTest_InvalidOffset) {
+    WriteTestFile("a", 1);
+
+    Assert::ExpectException<JSINativeException>([this] {
+      uint32_t badOffset = 3;
+      std::shared_ptr<Buffer> buffer =
+          MakeMemoryMappedBuffer(m_testFileName.c_str(), badOffset);
+    });
+  }
+};
```

Comment:
(nit) would be nice to place it into namespace per good C++ coding practice

Reply:
It looks like none of our unit test code is in a namespace. I'll send a separate PR and make said change to all of them.

---
In reply to: [321853819](https://github.com/microsoft/react-native-windows/pull/3059#discussion_r321853819) [](ancestors = 321853819)